### PR TITLE
Add activity_task_received metric and validate minimum number of activity task pollers

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: core/src/worker/activities.rs [hunk 1]
```diff
@@ -374,7 +375,21 @@ impl WorkerActivityTasks {
     }
 
     /// Called when there is a new act task about to be bubbled up out of the manager
-    fn about_to_issue_task(&self, task: PermittedTqResp) -> ActivityTask {
+    fn about_to_issue_task(&self, task: PermittedTqResp, is_eager: bool) -> ActivityTask {
+        if let Some(ref act_type) = task.resp.activity_type {
+            if let Some(ref wf_type) = task.resp.workflow_type {
+                self.metrics
+                    .with_new_attrs([
```
Comment (cretz) [inline core/src/worker/activities.rs:382]: I assume task queue and namespace is already present?

**Review by cretz — APPROVED** [review]
Each language's SDK features harness will also need a way to expose its worker's metrics. In Java and Go this can be done w/ raw capturing using metric abstractions. In TS maybe you can have a Otel collector on a local URL, but in Python this is tricky due to proto issues right now.

File: core/src/worker/activities.rs [hunk 2]
```diff
@@ -374,7 +375,21 @@ impl WorkerActivityTasks {
     }
 
     /// Called when there is a new act task about to be bubbled up out of the manager
-    fn about_to_issue_task(&self, task: PermittedTqResp) -> ActivityTask {
+    fn about_to_issue_task(&self, task: PermittedTqResp, is_eager: bool) -> ActivityTask {
+        if let Some(ref act_type) = task.resp.activity_type {
+            if let Some(ref wf_type) = task.resp.workflow_type {
```
Comment (Sushisource) [inline core/src/worker/activities.rs:380]: You should be able to combine these with the new let chains feature! https://rust-lang.github.io/rfcs/2497-if-let-chains.html 

`if let Some(ref act_type) = task.resp.activity_type && let Some(ref wf_type) = task.resp.workflow_type {`

File: core/src/worker/activities.rs (hunk 1 above)
Comment (Sushisource) [reply core/src/worker/activities.rs:382]: Yeah that's part of the metrics context

**Review by Sushisource — APPROVED** [review]

**Comment by bergundy** [comment]
> Each language's SDK features harness will also need a way to expose its worker's metrics. In Java and Go this can be done w/ raw capturing using metric abstractions. In TS maybe you can have a Otel collector on a local URL, but in Python this is tricky due to proto issues right now.

We can scrape the prom endpoint.

File: core/src/worker/activities.rs (hunk 2 above)
Comment (bergundy) [reply core/src/worker/activities.rs:380]: I didn't know that was stable, thanks!

Comment (bergundy) [reply core/src/worker/activities.rs:380]: It's unstable:

https://github.com/rust-lang/rust/issues/53667

Comment (Sushisource) [reply core/src/worker/activities.rs:380]: Bleh, they stabilized part of it but I have no idea which part lol because every time I try to use it I fail

**Pull request merged**